CMakeLists.txt

```diff
@@ -302,6 +302,11 @@ include (cmake/find_prometheus.cmake)
 include (cmake/find_tiflash_proxy.cmake)
 include (cmake/find_xxhash.cmake)
 
+if (USE_QPL AND OS_LINUX AND ARCH_AMD64 AND TIFLASH_ENABLE_AVX512_SUPPORT)
+    option (ENABLE_QPL "Enable Intel Query Processing Library" ON)
+    include (cmake/find_qpl.cmake)
+endif()
```

Comment:
```suggestion
if (OS_LINUX AND ARCH_AMD64 AND TIFLASH_ENABLE_AVX512_SUPPORT)
    set (ENABLE_QPL_DEFAULT ON)
else ()
    set (ENABLE_QPL_DEFAULT ON)
endif()
option (ENABLE_QPL "Enable Intel Query Processing Library" ${ENABLE_QPL_DEFAULT})

if (ENABLE_QPL)
    if (OS_LINUX AND ARCH_AMD64 AND TIFLASH_ENABLE_AVX512_SUPPORT)
        include (cmake/find_qpl.cmake)
    else ()
        message (FATAL_ERROR "`ENABLE_QPL` is not supported on this platform")
    endif ()
endif ()
```

Reply:
Line 308 should be "set (ENABLE_QPL_DEFAULT OFF)", right?
The other parts are good.

Reply:
It is, I made a modification minutes after my original change. Maybe your page is outdated.

But by looking again I just realized two things:
1. This logic could be further simplified.
2. For developers who need to build tiflash, we don't want to enforce using qpl for them, i.e., require asm_nasm installed in the system. So maybe it's more appropriate to default this option to false. Consequently, we'll explicitly enable this option in our official build, which will be done in other scripts. I'll update the script later.

Reply:
Looks good to me.